### PR TITLE
Fixes tensorrt cache being regenerated on path change

### PR DESCRIPTION
**Pull request by fran6co**
Onnxruntime uses the path as a hash for the tensorrt cache, passing a binary avoids this and works as intended.

Fixes https://github.com/triton-inference-server/server/issues/4587

### CODE REVIEW

**Comment by GuanLuo** [comment]
@pranavsharma can you or someone familiar with TRT EP take a look at this PR? The change looks fine to me and just want to double check if the cache behavior will be changed as intended.

**Comment by pranavsharma** [comment]
@stevenlix - any comments? Is this how we avoid TRT cache regeneration?

**Comment by stevenlix** [comment]
Yes. ORT-TRT uses ModelMetadefIdGenerator function to generate engine id. If the model passed in has a path, the id will be generated from the path string (https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/framework/execution_provider.cc#L151). To avoid the dependency on path, one can pass model binary to the inference session.

**Comment by cnegron-nv** [comment]
@pranavsharma Can you give a final review to this PR?

**Review by pranavsharma — DISMISSED** [review]

**Review by tanmayv25 — APPROVED** [review]

File: src/onnxruntime_loader.cc
Comment (Tabrizian) [inline src/onnxruntime_loader.cc:195]: Can you add a comment mentioning that the reason to pass the ONNX file as a binary is to avoid the TRT cache from being regenerated?

**Review by Tabrizian — APPROVED** [review]

**Review by rmccorm4 — APPROVED** [review]

**Comment by Tabrizian** [comment]
Hi @fran6co, @pranavsharma, @stevenlix 

We are observing a failure in the CI as a result of this change. Looks like for models that expect additional weight files the relative paths do not work properly anymore. The test fails with the error below:

```
Deserialize tensor embeddings.20.weight failed.open file "./embeddings.20.weight" failed: No such file or directory |
```

Do you have a fix in mind for this? Otherwise, I think we need to revert this change.

**Comment by pranavsharma** [comment]
I've reached out to @stevenlix. When is the latest you want this fix? If it's blocking the release feel free to revert it.

**Comment by Tabrizian** [comment]
Thanks for the quick response! We are code freezing by the end of next week but I think it would be great if it is possible to have the fix ready for review by Wednesday next so that we can test it before we are code frozen.

**Comment by pranavsharma** [comment]
> Hi @fran6co, @pranavsharma, @stevenlix
> 
> We are observing a failure in the CI as a result of this change. Looks like for models that expect additional weight files the relative paths do not work properly anymore. The test fails with the error below:
> 
> ```
> Deserialize tensor embeddings.20.weight failed.open file "./embeddings.20.weight" failed: No such file or directory |
> ```
> 
> Do you have a fix in mind for this? Otherwise, I think we need to revert this change.

fyi @jywu-msft

**Comment by stevenlix** [comment]
@skottmckay, one solution is to not use absolute model path in https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/framework/execution_provider.cc#L151

**Comment by pranavsharma** [comment]
Let's revert this at this point because using byte stream won't work with external weight files. So, this PR *will* fail.

**Comment by bamdadd** [comment]
Hi, @Tabrizian Do you plan to revert the revert anytime soon?

**Comment by Tabrizian** [comment]
The underlying issue has not been fixed yet so I don't think there is any plan to revert this change.

CC @pranavsharma

**Pull request merged**
